dpsim-models/src/DP/DP_Ph1_SynchronGenerator5bOrderVBR.cpp

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2017-2021 Institute for Automation of Complex Power Systems,
+ *                     EONERC, RWTH Aachen University
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *********************************************************************************/
+
+#include <dpsim-models/DP/DP_Ph1_SynchronGenerator5bOrderVBR.h>
+
+using namespace CPS;
+
+DP::Ph1::SynchronGenerator5bOrderVBR::SynchronGenerator5bOrderVBR
+    (const String & uid, const String & name, Logger::Level logLevel)
+	: ReducedOrderSynchronGeneratorVBR(uid, name, logLevel),
+	mEdq_t(mAttributes->create<Matrix>("Edq_t")),
+	mEdq_s(mAttributes->create<Matrix>("Edq_s")) {
+
+	//
+	mSGOrder = SGOrder::SG5bOrder;
+
+	// model DPecific variables
+	**mEdq_t = Matrix::Zero(2,1);
+	**mEdq_s = Matrix::Zero(2,1);
+	mEh_s = Matrix::Zero(2,1);
+	mEh_t = Matrix::Zero(2,1);
+}
+
+DP::Ph1::SynchronGenerator5bOrderVBR::SynchronGenerator5bOrderVBR
+	(const String & name, Logger::Level logLevel)
+	: SynchronGenerator5bOrderVBR(name, name, logLevel) {
+}
+
+void DP::Ph1::SynchronGenerator5bOrderVBR::specificInitialization() {
+
+	// initial voltage behind the transient reactance in the dq reference frame
+	(**mEdq_t)(0,0) = 0.0;
+	(**mEdq_t)(1,0) = (1 - mTaa / mTd0_t) * **mEf - (mLd - mLd_t - mYd) * (**mIdq)(0,0);
+
+	// initial dq behind the subtransient reactance in the dq reference frame
+	(**mEdq_s)(0,0) = (**mVdq)(0,0) - (mLq_s) * (**mIdq)(1,0);
+	(**mEdq_s)(1,0) = (**mVdq)(1,0) + (mLd_s) * (**mIdq)(0,0);
+
+	mSLog->info(
+		"\n--- Model DPecific initialization  ---"
+		"\nSG model: 5th order type 2"
+		"\nInitial Ed_t (per unit): {:f}"
+		"\nInitial Eq_t (per unit): {:f}"
+		"\nInitial Ed_s (per unit): {:f}"
+		"\nInitial Eq_s (per unit): {:f}"
+		"\n--- Model DPecific initialization finished ---",
+		(**mEdq_t)(0,0),
+		(**mEdq_t)(1,0),
+		(**mEdq_s)(0,0),
+		(**mEdq_s)(1,0)
+	);
+	mSLog->flush();
+}
+
+void DP::Ph1::SynchronGenerator5bOrderVBR::stepInPerUnit() {
+	// update DP-DQ transforms
+	mDomainInterface.updateDQToDPTransform(**mThetaMech, mSimTime);
+	mDomainInterface.updateDPToDQTransform(**mThetaMech, mSimTime);
+
+	if (mSimTime>0.0){
```

Comment:
Is `mSimTime>0.0` still required? E.g. [DP_Ph1_SynchronGenerator4OrderVBR](https://github.com/sogno-platform/dpsim/blob/b228bc46c0c7087291645715a22eab5aa0e9e74b/dpsim-models/src/DP/DP_Ph1_SynchronGenerator4OrderVBR.cpp#L54-L56) does not need it